agent/src/softwarecontaineragent.cpp

```diff
@@ -31,24 +33,32 @@ void SoftwareContainerAgent::triggerPreload()
 
 void SoftwareContainerAgent::deleteContainer(ContainerID containerID)
 {
-    bool valid = ((containerID < m_containers.size()) && (m_containers[containerID] != nullptr));
-    if (valid) {
-        m_containers[containerID] = nullptr;
-    } else {
-        log_error() << "Invalid container ID " << containerID;
-    }
+	bool valid = ((containerID < UINT32_MAX )
```

Comment:
Put the two identical checks for validity in a private method, or inside the "ID generator class" I proposed below, if that works out.